pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -901,6 +903,35 @@ public String getServerMetadata(
     return segmentsMetadata;
   }
 
+  @GET
+  @Path("segments/{tableName}/zkmetadata")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get the zookeeper metadata for all table segments",
+          notes = "Get the zookeeper metadata for all table segments")
+  public String getZookeeperMetadata(
+          @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+          @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr)
+          throws JsonProcessingException {
+    LOGGER.info("Received a request to fetch zookeeper metadata for all segments for table {}", tableName);
+    TableType tableType = Constants.validateTableType(tableTypeStr);
+
+    List<String> tableNamesWithType =
+            ResourceUtils.getExistingTableNamesWithType(_pinotHelixResourceManager, tableName, tableType, LOGGER);
+    Map<String, JsonNode> tableTypeToSegmentMetadata = new HashMap<>();
```

Comment:
Suggest putting the table name as the key, or just allow querying a single table (same as server metadata)

Reply:
I tried to keep it in parity with how we return for other requests using OFFLINE and REALTIME as key, e.g. for list of segments. Let me know if we are planning to return `tableNameWithType` instead going forward.

Reply:
Seems we have different return type for different API. Since this one is for metadata, maybe keep it the same as server metadata API? Basically only return the metadata for the first table if table type is not provided

Reply:
Just confirming if table type is not provided, then we return only OFFLINE table metadata right?

Reply:
Made it to return one table info